src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = [];
+
+  actions.forEach(action => {
+    if (action.type === 'addProperties') {
+      const add = {
+        ...(copy.length
+          ? copy[copy.length - 1]
+          : state),
+        ...action.extraData,
+      };
+
+      copy.push(add);
```

Comment:
you can place `copy.push` at the very end of `for` loop and use one variable `stateCopy` to adjust it